src/components/AnonymousChat/child/ChatItemTargetText.tsx

```diff
@@ -39,41 +49,58 @@ const ChatItemTargetText = ({
   messageType,
   data
 }: ChatItemMyTextProps) => {
-  const {setReplyPreview, onContextMenuPressed} = useMessageHook();
+  const {
+    setReplyPreview,
+    onContextMenuPressed,
+    pulseAnimation,
+    animatedBubbleStyle,
+    animatedPulseStyle
+  } = useMessageHook();
+  const scrollContext = React.useContext(ScrollContext);
+  const swipeableRef = React.useRef<Swipeable>(null);
 
-  const swipeableRef = React.useRef<Swipeable | null>(null);
-  const bubblePosition = useSharedValue(0);
+  React.useEffect(() => {
+    if (scrollContext?.selectedMessageId === data?.id) {
+      pulseAnimation.value = withSequence(
+        withDelay(300, withTiming(1.1, {duration: 200})),
+        withTiming(1, {duration: 200})
+      );
+      scrollContext?.setSelectedMessageId(null);
+    }
+  }, [scrollContext?.selectedMessageId]);
 
-  const isReply = messageType === 'reply';
-  const isReplyPrompt = messageType === 'reply_prompt';
+  const isReply = messageType === MESSAGE_TYPE_REPLY;
+  const isReplyPrompt = messageType === MESSAGE_TYPE_REPLY_PROMPT;
   const isShowUserInfo = !isContinuous || isReplyPrompt || isReply;
 
-  const animatedBubbleStyle = useAnimatedStyle(() => ({
-    transform: [{translateX: bubblePosition.value}]
-  }));
-
-  const contextMenuActions = [
-    {title: 'Reply', systemIcon: 'arrow.turn.up.left'},
-    {title: 'Copy Message', systemIcon: 'square.on.square'}
+  const contextMenuActions: ContextMenuAction[] = [
+    {title: CONTEXT_MENU_REPLY, systemIcon: 'arrow.turn.up.left'}
   ];
 
+  if (messageType !== MESSAGE_TYPE_DELETED) {
+    contextMenuActions.push({title: CONTEXT_MENU_COPY, systemIcon: 'square.on.square'});
+  }
+
   const renderAvatar = React.useCallback(() => {
     if (!isShowUserInfo) return <View style={styles.avatar} />;
-    return <View style={styles.mr8}>{avatar}</View>;
-  }, []);
+    return <View style={styles.mrBuble}>{avatar}</View>;
+  }, [isShowUserInfo, avatar]);
 
-  const onSwipeToReply = (direction) => {
-    if (direction === 'right') return;
-    if (swipeableRef.current) swipeableRef.current?.close();
-    setReplyPreview({
-      username,
-      time,
-      message,
-      messageId: data?.id,
-      chatType,
-      messageType: 'regular'
-    });
-  };
+  const onSwipeToReply = React.useCallback(
+    (direction: 'left' | 'right') => {
+      if (direction === 'right') return;
+      if (swipeableRef.current) swipeableRef.current?.close();
+      setReplyPreview({
+        id: data?.id ?? data?.message?.id,
+        user: {username},
+        message,
+        message_type: messageType as MessageType,
+        updated_at: time,
+        chatType
+      });
+    },
+    [data, username, time, message, messageType, chatType]
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useMessageHook` hook is used here, but it's not clear if the hook is properly handling dependencies. It seems to be returning a set of functions and values that are used in this component, but without seeing the implementation of the hook, it's hard to say if there might be unnecessary re-renders or stale data issues. Please ensure that all dependencies are correctly handled in the `useMessageHook` hook.

Also, in line 63, you're checking if `scrollContext?.selectedMessageId === data?.id`. If `data` is null or undefined, this condition will always be false. You should add a check for `data` before this condition.

```diff
  React.useEffect(() => {
+    if (!data) return;
    if (scrollContext?.selectedMessageId === data?.id) {
      pulseAnimation.value = withSequence(
        withDelay(300, withTiming(1.1, {duration: 200})),
        withTiming(1, {duration: 200})
      );
      scrollContext?.setSelectedMessageId(null);
    }
  }, [scrollContext?.selectedMessageId]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->